lib/chef/resource_reporter.rb

```diff
@@ -198,6 +198,17 @@ def resource_failed(new_resource, action, exception)
     def resource_completed(new_resource)
       if @pending_update && !nested_resource?(new_resource)
         @pending_update.finish
+
+        # Verify if the resource has sensitive data
+        # and create a new blank resource with only
+        # the name so we can report it back without
+        # sensitive data
+        if @pending_update.new_resource.sensitive
+          klass = @pending_update.new_resource.class
+          resource_name = @pending_update.new_resource.name
+          @pending_update.new_resource = klass.new(resource_name)
```

Comment:
This leaves no way to know if a resource was actually blank or sensitive:

```ruby
package 'foo'

# vs

package 'foo' do
  sensitive true
end
```

I don't really need to worry much about y'alls commercial features so I suppose it's up to you if you think that's a problem. I will also point out that we already have a method for fixing this, `Chef::Resource#state_for_resource_reporter`.